### DISCUSSION TITLE
standalone installation of epoptes

### DISCUSSION THREAD

**Discussion by pappaspanos** [epoptes/epoptes#113] [Q&A]
Hi. I am writing to you for a problem with epoptes. We have a school lab with 11 PCs, Intel E52xx with 1GB Ram, except 1 with 3GB. Because they are very old and we don't have a gigabit switch but 3-4 switches, the one connected to another, we decided to install Lubuntu 18.04 LTS in every computer adn we can say that they work quickly enough. The PC with 3GB Ram is the theacher's one and here we want to run epoptes. We followed the guide here -https://epoptes.org/installation/ - but we cannot see any client. I have to say that every PC has hostname like 3lab-pc1, 3lab-pc2 etc. and there is a root user with username USER. Furthermore, because we don't have static IPs I did those steps in every PC

If this works, then put that name in /etc/default/epoptes-client. On each client, run:
sudo nano /etc/default/epoptes-client
and replace the line that says
#SERVER=server
with
SERVER=epoptes-server.local

After you do (2) or (3), you should also fetch the epoptes certificate on each client:
sudo epoptes-client -c
And then you should restart the client:
sudo reboot

and i used the name 3lab-pc1.local for server.
can you give me any help? Thanks

**Comment by alkisg** [comment]
Hi Pano,

>  SERVER=epoptes-server.local
> and i used the name 3lab-pc1.local for server.

These two names are supposed to match. So put `SERVER=3lab-pc1.local`, and then run `epoptes-client -c`, and then reboot.

**Comment by pappaspanos** [comment]
Hi Alkis. Thanks for your reply. I have done those steps you described but no result. Beside that I removed and reinstalled epoptes using the stable repository, as it is described here - https://epoptes.org/documentation/ppa/ - but I steel cannot see any client in server. Is there anything to do in server side, beside from installing epoptes and allowing users to launch the GUI  with the command gpasswd -a username epoptes???

**Comment by alkisg** [comment]
Hi Panos,

no you don't need to do anything else in the server. You can verify that the epoptes service is running by issuing `systemctl status epoptes`. And you can verify that the certificates are fine by running `ls -l /etc/epoptes`.

The problem is in the clients. When you fetched the certificate, did it report success? What happens if you run `ls -l /etc/epoptes` in the clients, does server.crt have the same size as the one in the server? What happens if you run `/usr/sbin/epoptes-client`, does it mention that it properly connects? Do you then see it in the epoptes GUI on the server?

Btw, Greek schools have better support in http://alkisg.mysch.gr/steki/index.php?board=67.0.

**Comment by pappaspanos** [comment]
Hi Alkis. I have done all those you mentioned above and here are the results:
- when i'm running _systemctl status epoptes_ it tells me it is loaded but Active:Failed
- when is rum the command _ls -l /etc/epoptes_ the _server.crt_ has the same size as the one in the server, 1939 bytes
- when i run the command _sudo /usr/sbin/epoptes-client_ its seems to be ok, the last two line are
DONE
Successfully fetched certificate from 3lab-pc1.local:789

I cannot understand where is the problem???
i am attaching some printscreens in order to be more specific and maybe to guide me to find it.
![εικόνα](https://user-images.githubusercontent.com/61320294/75155797-7495e180-5719-11ea-992f-a5e43c5a0e32.png)
![εικόνα](https://user-images.githubusercontent.com/61320294/75155851-91cab000-5719-11ea-9cfc-b120643d69f9.png)
![εικόνα](https://user-images.githubusercontent.com/61320294/75155903-aad36100-5719-11ea-8acc-68a5e406d380.png)
Do you want me to move the conversation to the forum you mention?

**Comment by alkisg** [comment]
Hi Pano. In the clients, you're not supposed to have epoptes installed. You only need epoptes-client there. So, try these:

```shell
sudo -i
apt install epoptes-client
apt purge --auto-remove epoptes
epoptes-client -c
reboot
```

**Comment by alkisg** [comment]
Solved in IRC, closing.

**Comment by pappaspanos** [comment]
Alkis sorry to write so late, thanks a lot for your help. I thinks it's worth to mention that you solved the problem by declaring static IP to server and then putting that IP and the server name to /etc/hosts. Thanks again.

**Comment by alkisg** [comment]
Yeah the main problem in your system was that while `ping epoptes-pc.local` worked, `getent hosts epoptes-pc.local` didn't work.
That's a misconfiguration in your system, but we didn't take the time to see what caused that misconfiguration as it's not related to epoptes, so we just worked around it with the static IP.
